### PR TITLE
Adds unit spec for useMyHealthAccessGuard

### PR DESCRIPTION
**Pull request by radavis**
## Summary

- Adds unit spec for useMyHealthAccessGuard

## Related issue(s)

department-of-veterans-affairs/va.gov-team#84911

## Testing done

- localhost, CI

## Screenshots

No UI changes

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/platform/mhv/tests/hooks/useMyHealthAccessGuard.unit.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  return <h1>Health Tool</h1>;
+};
+
+const setup = (initialState = stateFn()) =>
+  renderWithStoreAndRouter(<Component />, {
+    initialState,
+  });
+
+describe('useMyHealthAccessGuard', () => {
+  let originalLocation;
```
Comment (carlosfelix2) [inline src/platform/mhv/tests/hooks/useMyHealthAccessGuard.unit.spec.jsx:28]: Is this safe? Do you really need to set the original location at the end? I feel you can just set the spy before the test and nothing else.

File: src/platform/mhv/tests/hooks/useMyHealthAccessGuard.unit.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    window.location = {
+      replace: sinon.spy(),
+    };
+  });
+
+  afterEach(() => {
+    window.location = originalLocation;
+  });
+
+  describe('with valid MHV account', () => {
```
Comment (carlosfelix2) [inline src/platform/mhv/tests/hooks/useMyHealthAccessGuard.unit.spec.jsx:42]: Why have a describe and it? Just have the it.

File: src/platform/mhv/tests/hooks/useMyHealthAccessGuard.unit.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+  describe('with multiple MHV accounts', () => {
+    it('renders the component', () => {
+      const initialState = stateFn({ mhvAccountState: 'MULTIPLE' });
+      const { getByRole } = setup(initialState);
+      getByRole('heading', { name: 'Health Tool' });
+    });
+  });
+
+  describe('without a valid MHV account', async () => {
```
Comment (carlosfelix2) [inline src/platform/mhv/tests/hooks/useMyHealthAccessGuard.unit.spec.jsx:57]: This one has not `it`. Maybe be consistent on how these are organized.

File: src/platform/mhv/tests/hooks/useMyHealthAccessGuard.unit.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+const stateFn = ({ mhvAccountState = 'OK' } = {}) => ({
+  user: {
+    profile: {
+      mhvAccountState,
+    },
+  },
+});
+
+const Component = () => {
+  useMyHealthAccessGuard();
```
Comment (carlosfelix2) [inline src/platform/mhv/tests/hooks/useMyHealthAccessGuard.unit.spec.jsx:18]: Why not just inline this? You don't use Component anywhere else.

**Review by carlosfelix2 — DISMISSED** [review]
With comments

**Comment by carlosfelix2** [comment]
Also, cleanup the PR description like the test section.

File: src/platform/mhv/tests/hooks/useMyHealthAccessGuard.unit.spec.jsx (hunk 1 above)
Comment (radavis) [reply src/platform/mhv/tests/hooks/useMyHealthAccessGuard.unit.spec.jsx:28]: This is a fairly common pattern in `vets-website` for spying on `window.location.replace`.

File: src/platform/mhv/tests/hooks/useMyHealthAccessGuard.unit.spec.jsx (hunk 2 above)
Comment (radavis) [reply src/platform/mhv/tests/hooks/useMyHealthAccessGuard.unit.spec.jsx:42]: The `describe` blocks detail the unit under test and the conditions. The 'it' blocks describe the outcome.

**Review by jzucadi — APPROVED** [review]

**Pull request merged**